CRM/Core/SmartyCompatibility.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ * Fix for bug CRM-392. Not sure if this is the best fix or it will impact
+ * other similar PEAR packages. doubt it
+ */
+if (!class_exists('Smarty')) {
+  if (defined('CIVICRM_SMARTY3')) {
+    global $civicrm_root;
+    require_once $civicrm_root . '/ext/smarty3/vendor/autoload.php';
```

Comment:
Also, shouldn't it be like
`CRM_Utils_File::addTrailingSlash($civicrm_root) . 'ext...`